internal/datastore/crdb/watch.go

```diff
@@ -24,6 +24,7 @@ import (
 const (
 	queryChangefeed       = "EXPERIMENTAL CHANGEFEED FOR %s WITH updated, cursor = '%s', resolved = '%s', min_checkpoint_frequency = '0';"
 	queryChangefeedPreV22 = "EXPERIMENTAL CHANGEFEED FOR %s WITH updated, cursor = '%s', resolved = '%s';"
+	watchConnectTimeout   = 1 * time.Second
```

Comment:
I'd suggest making this more conservative, and tunable. The code moves from no timeout at all to 1 second, and that could backfire in situations where a CRDB cluster is at capacity

Reply:
That's actually kind of the point - it would fail fast and then they'd retry. It only applies on the initial connect

Reply:
Made configurable